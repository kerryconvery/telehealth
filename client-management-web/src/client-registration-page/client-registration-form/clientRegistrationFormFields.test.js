import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import ClientRegistrationFormFields from './clientRegistrationFormFields';


describe('When validating client registration fields', () => {
  it('should return validation errors when required fields are missing', () => {
    const client = {};

    const validationErrors = ClientRegistrationFormFields.validateFields(client);
    
    expect(validationErrors.title).toBeDefined();
    expect(validationErrors.firstName).toBeDefined();
    expect(validationErrors.lastName).toBeDefined();
    expect(validationErrors.phoneNumber).toBeDefined();
  });

  it('should return no validation errors when all required fields are present', () => {
    const client = {
      title: 'Mr',
      firstName: 'Client',
      lastName: 'One',
      phoneNumber: '12122'
    };

    const validationErrors = ClientRegistrationFormFields.validateFields(client);
    
    expect(validationErrors).toEqual({});
  });
})

describe('When there are validation errros', () => {
  const baseProps = {
    client: {},
    validationErrors: {},
    onChange: jest.fn(),
  }

  it('should display the errors against their respective controls', () => {
    const validationErrors = ClientRegistrationFormFields.validateFields({});

    const props = {
      ...baseProps,
      validationErrors,
    }
    
    const component = shallow(<ClientRegistrationFormFields { ...props } />)

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should not display errors when there are not validation errors', () => {    
    const component = shallow(<ClientRegistrationFormFields { ...baseProps } />)

    expect(shallowToJson(component)).toMatchSnapshot();
  });
})