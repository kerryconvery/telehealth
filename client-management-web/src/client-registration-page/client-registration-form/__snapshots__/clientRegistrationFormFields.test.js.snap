// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When there are validation errros should display the errors against their respective controls 1`] = `
<WithStyles(ForwardRef(FormControl))
  component="fieldset"
>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-shortFormControl-1"
  >
    <WithStyles(ForwardRef(TextField))
      defaultValue=""
      error={true}
      helperText="Please choose a title"
      id="title"
      label="Title"
      onChange={[Function]}
      select={true}
    >
      <WithStyles(ForwardRef(MenuItem))
        value="Mr"
      >
        Mr
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="Mrs"
      >
        Mrs
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="Miss"
      >
        Miss
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="Ms"
      >
        Ms
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(TextField))>
  </WithStyles(ForwardRef(FormControl))>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-longFormControl-2"
  >
    <WithStyles(ForwardRef(TextField))
      error={true}
      helperText="Please enter a first name"
      id="first-name"
      label="First name"
      onChange={[Function]}
    />
  </WithStyles(ForwardRef(FormControl))>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-longFormControl-2"
  >
    <WithStyles(ForwardRef(TextField))
      error={true}
      helperText="Please enter a last name"
      id="last-name"
      label="Last name"
      onChange={[Function]}
    />
  </WithStyles(ForwardRef(FormControl))>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-shortFormControl-1"
  >
    <WithStyles(ForwardRef(TextField))
      error={true}
      helperText="Please enter a phone number"
      id="phone-number"
      label="Phone number"
      onChange={[Function]}
    />
  </WithStyles(ForwardRef(FormControl))>
</WithStyles(ForwardRef(FormControl))>
`;

exports[`When there are validation errros should not display errors when there are not validation errors 1`] = `
<WithStyles(ForwardRef(FormControl))
  component="fieldset"
>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-shortFormControl-1"
  >
    <WithStyles(ForwardRef(TextField))
      defaultValue=""
      error={false}
      helperText="Required"
      id="title"
      label="Title"
      onChange={[Function]}
      select={true}
    >
      <WithStyles(ForwardRef(MenuItem))
        value="Mr"
      >
        Mr
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="Mrs"
      >
        Mrs
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="Miss"
      >
        Miss
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="Ms"
      >
        Ms
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(TextField))>
  </WithStyles(ForwardRef(FormControl))>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-longFormControl-2"
  >
    <WithStyles(ForwardRef(TextField))
      error={false}
      helperText="Required"
      id="first-name"
      label="First name"
      onChange={[Function]}
    />
  </WithStyles(ForwardRef(FormControl))>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-longFormControl-2"
  >
    <WithStyles(ForwardRef(TextField))
      error={false}
      helperText="Required"
      id="last-name"
      label="Last name"
      onChange={[Function]}
    />
  </WithStyles(ForwardRef(FormControl))>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-shortFormControl-1"
  >
    <WithStyles(ForwardRef(TextField))
      error={false}
      helperText="Required"
      id="phone-number"
      label="Phone number"
      onChange={[Function]}
    />
  </WithStyles(ForwardRef(FormControl))>
</WithStyles(ForwardRef(FormControl))>
`;
